docs/source/checks/nlp/model_evaluation/plot_metadata_segments_performance.py

```diff
@@ -0,0 +1,107 @@
+# -*- coding: utf-8 -*-
+"""
+Metadata Segments Performance
+*************************
+
+This notebook provides an overview for using and understanding the metadata segment performance check.
+
+**Structure:**
+
+* `What is the purpose of the check? <#what-is-the-purpose-of-the-check>`__
+* `Automatically detecting weak segments <#automatically-detecting-weak-segments>`__
+* `Generate data & model <#generate-data-model>`__
+* `Run the check <#run-the-check>`__
+* `Define a condition <#define-a-condition>`__
+
+What is the purpose of the check?
+==================================
+
+The check is designed to help you easily identify the model's weakest segments based on the provided
+:func:`metadata <deepchecks.nlp.text_data.TextData.set_metadata>`. In addition,
+it enables to provide a sublist of the metadata columns, thus limiting the check to search in
+interesting subspaces.
+
+Automatically detecting weak segments
```

Comment:
I wonder if this explanation (and possibly also the "key parameters")
should be in some general concepts and then linked from all weak-segment checks.

I assume that you copy-pasted this from other segments checks, and if we take into account tabular, vision, nlp - there should be 5 "weak segment"  checks

(and btw maybe same for domain classifier if its more than twice in docs but not related directly here)

Reply:
maybe, but not now :)